pkg/collector/reconcile/config_replace.go

```diff
@@ -57,6 +57,11 @@ func ReplaceConfig(instance v1alpha1.OpenTelemetryCollector) (string, error) {
 		return "", getCfgPromErr
 	}
 
+	validateCfgPromErr := ta.ValidatePromConfig(promCfgMap, instance.Spec.TargetAllocator.Enabled, featuregate.EnableTargetAllocatorRewrite.IsEnabled())
```

Comment:
Is `ValidatePromConfig()` meant to be called every time `ta.ConfigToPromConfig()` is called? Put another way, is there ever a situation when we want to call one function without the other? I'm wondering the benefit of two separate functions.  Is it separation of concerns? Not making a breaking change to the `adapters` pkg API?

Reply:
Not making a breaking change and keeping current semantics of `ta.ConfigToPromConfig()` is part of it, although the separation of concerns is a nice bonus from that. Though overall there isn't any strong reason behind it, if we're ok breaking `ConfigToPromConfig`, we could simply make validation a private function and have `ConfigToPromConfig` call it.

Reply:
that sounds like a good followup for the future :)